src/main/java/me/jellysquid/mods/sodium/client/render/chunk/DefaultChunkRenderer.java

```diff
@@ -217,12 +224,32 @@ private GlTessellation prepareTessellation(CommandList commandList, RenderRegion
         return tessellation;
     }
 
+    private GlVertexAttributeBinding[] getBindingsForType() {
+        if(this.vertexType == ChunkMeshFormats.COMPACT) {
+            GlVertexFormat<ChunkMeshAttribute> compactFormat = (GlVertexFormat<ChunkMeshAttribute>)this.vertexFormat;
+            return new GlVertexAttributeBinding[] {
+                    new GlVertexAttributeBinding(ChunkShaderBindingPoints.ATTRIBUTE_PACKED_DATA,
+                            compactFormat.getAttribute(ChunkMeshAttribute.VERTEX_DATA))
+            };
+        } else if(this.vertexType == ChunkMeshFormats.VANILLA_LIKE) {
+            GlVertexFormat<VanillaLikeChunkMeshAttribute> vanillaFormat = (GlVertexFormat<VanillaLikeChunkMeshAttribute>)this.vertexFormat;
+            return new GlVertexAttributeBinding[] {
+                    new GlVertexAttributeBinding(ChunkShaderBindingPoints.ATTRIBUTE_POSITION,
+                            vanillaFormat.getAttribute(VanillaLikeChunkMeshAttribute.POSITION)),
+                    new GlVertexAttributeBinding(ChunkShaderBindingPoints.ATTRIBUTE_COLOR,
+                            vanillaFormat.getAttribute(VanillaLikeChunkMeshAttribute.COLOR)),
+                    new GlVertexAttributeBinding(ChunkShaderBindingPoints.ATTRIBUTE_TEXTURE_UV,
+                            vanillaFormat.getAttribute(VanillaLikeChunkMeshAttribute.TEXTURE_UV)),
+                    new GlVertexAttributeBinding(ChunkShaderBindingPoints.ATTRIBUTE_DRAW_PARAMS_LIGHT,
+                            vanillaFormat.getAttribute(VanillaLikeChunkMeshAttribute.DRAW_PARAMS_LIGHT)),
+            };
+        } else
+            return null; // assume Oculus/Iris will take over
```

Comment:
this section needs correct formatting: spaces between `if` and `(`, curly braces for the else block